### PR TITLE
Enable re-using existing websocket instance

### PR DESCRIPTION
**Pull request by RafalWilinski**
In this PR:
- rename `socket.io` (add with `esm_` prefix) to prevent event name clash with existing socket.io instance
- add an ability to supply optional `websocket` parameter to config which accepts existing socket.io instance
- bump socket.io to 1.5.1 and remove security vulnerabilities
- make port configurable (to make it possible using with existing socket.io)
- Add fancy badges 💯 
- Add tighter integration with bithound
- Add eslint as test step

Todo:
- [x] Test
- [x] Add tests
- [x] Extend documentation
- [x] Provide example

cc @mattiaerre

### CODE REVIEW

**Comment by mattiaerre** [comment]
Hi @RafalWilinski happy to have a look at this PR when ready to be reviewed.
keep me posted.
thanks,

**Comment by RafalWilinski** [comment]
@mattiaerre Done. 

Besides re-using existing web socket instance I've also added couple of 3rd integrations.

File: examples/index.js [hunk 1]
```diff
@@ -1,11 +1,23 @@
 /* eslint no-console: "off" */
+const socketIoPort = 2222;
 
 const express = require('express');
 
+// This is optional. If your server uses socket.io already, pass it to config as `webserver` along with it's port.
+const socketio = require('socket.io')(socketIoPort);
```
Comment (mattiaerre) [inline examples/index.js]: it is not possible to get the port that socket.io uses internally from the socket.io instance? I am trying to find out if that's possible 😉

File: package.json [hunk 2]
```diff
@@ -44,20 +44,28 @@
...
-    "prepublish": "npm run snyk-protect"
+    "prepublish": "npm run snyk-protect",
+    "example": "cd examples && npm start",
+    "eslint": "eslint . --fix"
   },
   "devDependencies": {
+    "bithound": "^1.7.0",
     "chai": "^3.5.0",
+    "eslint": "^3.9.1",
+    "eslint-config-airbnb": "^12.0.0",
```
Comment (mattiaerre) [inline package.json]: I ❤️ it!

File: src/helpers/on-headers-listener.js [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
   const category = Math.floor(statusCode / 100);
 
   spans.forEach((span) => {
     const last = span.responses[span.responses.length - 1];
-    if (last !== undefined && last.timestamp / 1000 + span.interval > Date.now() / 1000) {
-      last[category]++;
-      last.count++;
-      last.mean = last.mean + ((responseTime - last.mean) / last.count);
+    if (last !== undefined && (last.timestamp / 1000) + span.interval > Date.now() / 1000) {
+      last[category] += 1;
```
Comment (mattiaerre) [inline src/helpers/on-headers-listener.js]: good stuff !?!

File: src/middleware-wrapper.js [hunk 4]
```diff
@@ -12,17 +12,18 @@ const middlewareWrapper = (config) => {
     fs.readFileSync(path.join(__dirname, '/public/index.html'))
       .toString()
       .replace(/{{title}}/g, config.title)
+      .replace(/{{port}}/g, config.port)
```
Comment (mattiaerre) [inline src/middleware-wrapper.js]: so this is the port of the socket.io instance or the express server? I am confused 😅

File: src/public/index.html [hunk 5]
```diff
@@ -77,6 +77,7 @@ <h6 class="status-code status-code-5xx">5xx</h6>
   </div>
 </div>
 <script>
+  var port = {{port}};
```
Comment (mattiaerre) [inline src/public/index.html]: I would normally save this info inside a `data-*` attribute and read it inside the script w/ out polluting the global scope but ... not a major one

File: src/public/javascripts/app.js [hunk 6]
```diff
@@ -7,7 +7,7 @@ Chart.defaults.global.elements.line.backgroundColor = "rgba(0,0,0,0)";
 Chart.defaults.global.elements.line.borderColor = "rgba(0,0,0,0.9)";
 Chart.defaults.global.elements.line.borderWidth = 2;
 
-var socket = io(location.protocol + '//' + location.hostname + ':' + location.port);
+var socket = io(location.protocol + '//' + location.hostname + ':' + (port ? port : location.port));
 var defaultSpan = 0;
```
Comment (mattiaerre) [inline src/public/javascripts/app.js]: as per my comment before here we should be able to get the port value saved against some `data-*` attribute

File: .eslintignore [hunk 7]
```diff
@@ -0,0 +1,3 @@
+coverage
+examples
+src/public
```
Comment (RafalWilinski) [inline .eslintignore]: Noted, fixed!

File: examples/index.js (hunk 1 above)
Comment (RafalWilinski) [inline examples/index.js]: Yea, I was also wondering about that but couldn't find anything elegant.

File: src/helpers/on-headers-listener.js (hunk 3 above)
Comment (RafalWilinski) [inline src/helpers/on-headers-listener.js]: Can you elaborate on that? 😂

File: src/middleware-wrapper.js (hunk 4 above)
Comment (RafalWilinski) [inline src/middleware-wrapper.js]: Port of socket.io server which can be "plugged" or bind to express server (accordingly to socket.io documentation)

File: src/helpers/on-headers-listener.js (hunk 3 above)
Comment (mattiaerre) [inline src/helpers/on-headers-listener.js]: I like the `+=` instead of `++` 👍

**Comment by mattiaerre** [comment]
added more comments, over to you @RafalWilinski

**Comment by mattiaerre** [comment]
hi @RafalWilinski let me know if something is not clear as I'd love to have this PR merged ASAP 😄

**Pull request merged**